Adding -H "Authorization:Basic " possibility

justonestep opened the pull request:
Just added the possibility to use the standard Basic Authorization header.

```PHP
base64_encode('ClientId,ClientSecret);
```

**Example Request**
```
curl -H "Authorization:Basic VGVzdENsaWVudCxUZXN0U2VjcmV0" https://api.mysite.com/token -d 'grant_type=client_credentials'
{"access_token":"03807cb390319329bdf6c777d4dfae9c0d3b3c35","expires_in":3600,"token_type":"Bearer","scope":null}
```

Spomky commented:
The correct encoding is base64_encode(ClientId:ClientSecret);

And why do not use PHP_AUTH_USER, PHP_AUTH_PW, and AUTH_TYPE?

bshaffer commented:
This library already supports this:

https://github.com/bshaffer/oauth2-server-php/blob/develop/src/OAuth2/ClientAssertionType/HttpBasic.php#L100

Spomky commented:
This PR is a good idea (after the change above). 
This library should not rely only on the global variables. With some servers, they are not set and a check on the Authenticate header will fix this.

bshaffer commented:
@Spomky which web servers are you referring to which don't check this?

Spomky commented:
With PH-CGI or PH-FCGI for example.

bshaffer commented:
Okay, that seems reasonable.

@justonestep We would need a test case for this change, as well as changing the comment to a colon, as suggested by @Spomky. Also, if you could rebase it to be a single commit, that would be best.

Thank you for your contribution!

justonestep commented:
I changed the comment to a colon, added a test case and rebased it to an single commit
 (https://github.com/justonestep/oauth2-server-php/commit/baa30458abf258fe856ed9b4492ea619ef936431).
 Do I need to send a new pull request or is there another way to submit my changes?

bshaffer commented:
you could have kept the same PR open and just updated / rebased your branch. But you might as well submit a new PR now

justonestep commented:
I have no change to reopen this issue (button is deactivated "The feature/.. branch was force-pushed or recreated"). So I created a new [PR](https://github.com/bshaffer/oauth2-server-php/pull/501). Thanks!
